.travis.yml

```diff
@@ -14,12 +14,16 @@ jobs:
   - stage: test
     jdk: openjdk8
     script: ./gradlew check :test
-  # - stage: test                # Disabled due to SSL errors on Travis
-  #   jdk: openjdk7              # when downloading gradle
-  #   script: ./gradlew :test
   - stage: test
     jdk: oraclejdk9
     script: ./gradlew check :test
   - stage: test
     jdk: oraclejdk8
     script: ./gradlew check :test
+  - stage: test
+    jdk: openjdk7
+    script: ./gradlew :test
+    before_install: # Work around missing crypto in openjdk7
+    - sudo wget "https://bouncycastle.org/download/bcprov-ext-jdk15on-158.jar" -O "${JAVA_HOME}/jre/lib/ext/bcprov-ext-jdk15on-158.jar"
+    - sudo perl -pi.bak -e 's/^(security\.provider\.)([0-9]+)/$1.($2+1)/ge' /etc/java-7-openjdk/security/java.security
+    - echo "security.provider.1=org.bouncycastle.jce.provider.BouncyCastleProvider" | sudo tee -a /etc/java-7-openjdk/security/java.security
```

Comment:
What does this workaround do? Is this required due to SSL errors when downloading gradle?

Reply:
Yep! OpenJDK7 is missing the crypto to download the gradle packages securely, since oracle pulled the binaries a bit back. This downloads an implementation from [bouncycastle](https://www.bouncycastle.org/latest_releases.html) and changes config files to indicate it is available.